perma_web/perma/tasks.py

```diff
@@ -1837,7 +1837,15 @@ def add_metadata_to_existing_daily_item_files(file_ids):
             rate_limit_approaching = True
         if s3_is_overloaded or rate_limit_approaching:
             logger.warning(f"Skipped add metadata task for {file_id} (IA Item {ia_item.identifier}, File {link.guid}) due to rate limit.")
-            file_ids_to_retry.append(file_id)
+            retry = (
+                not settings.INTERNET_ARCHIVE_RETRY_FOR_RATELIMITING_LIMIT or
+                not previous_attempts or
+                (settings.INTERNET_ARCHIVE_RETRY_FOR_RATELIMITING_LIMIT > previous_attempts[file_id] + 1)
```

Comment:
Does this need to index `previous_attempts` with `str(file_id)` like you do elsewhere?

Reply:
Gulp: yes. Thanks for the careful proofreading. I clearly fired this off too quickly.

Reply:
np -- I was just trying to figure out why you weren't using the raw IDs as dictionary keys. I suppose that integer dict keys are probably a code smell.

Reply:
Ah you explained it in the commit message: "Dict key is a string (required by celery task format)"

Reply:
mypy's also helpful here for catching those sorts of things, assuming you defined the dict as having keys of type string.

Reply:
Yeah! I was passing in the integers initially, and couldn't figure out why I kept getting key errors... Hence that print statement you found, actually. It turned out that the keys of `previous_attempts` were strings, despite my having passed in ints.

I wasn't sure if it was better to explicitly convert it to a string once, and then use that var everywhere, or cast each time... My thought was, I was equally likely to use the wrong var as I was to forget to cast, so six of one half dozen of another...

Kind of a bummer the ints don't stick around, I'm guessing it's a `pickle` thing.